plugins/modules/system_high_availability_settings.py

```diff
@@ -0,0 +1,489 @@
+    for ident, desc in interfaces.items():
+        if sync_interface.lower() in (ident.lower(), desc.lower()):
+            config.set(ident, "synchronize_interface")
+            return
+    raise ValueError(
+        f"'{sync_interface}' is not a valid interface. "
+        + "If the interface exists, ensure it is enabled and also not virtual."
+    )
+
+
+def synchronize_peer_ip(config: OPNsenseModuleConfig, peer_ip: str) -> None:
+    """
+    Handler function for the synchronize_peer_ip setting
+    Args:
+        config (OPNsenseModuleConfig): The configuration for the opnsense firewall
+        peer_ip: PFsync will sync to this IP address.
+    """
+    if peer_ip:
+        if not validate_ipv4(peer_ip):
+            raise ValueError(
+                "Setting synchronize_peer_ip has to be a valid IPv4 address"
+            )
+        config.set(value=peer_ip, setting="synchronize_peer_ip")
+    elif not peer_ip and config.get("synchronize_peer_ip") is not None:
+        config.get("hasync").remove(config.get("synchronize_peer_ip"))
+
+
+def remote_system_synchronization(
+    config: OPNsenseModuleConfig,
+    remote_backup_url: Optional[str],
+    username: Optional[str],
+    password: Optional[str],
+) -> None:
+    """
+    Handler function for the settings synchronize_config_to_ip,
+    remote_system_username and remote_system_password.
+    Args:
+        config (OPNsenseModuleConfig): The configuration for the opnsense firewall
+        remote_backup_url (Optional[str]): Synchronize your settings to this URL
+        username (Optional[str]): Username for logging in to the remote firewall
+        password (Optional[str]): Password for logging in to the remote firewall
+    """
+    if any((remote_backup_url, username, password)):
+        config.set(value=remote_backup_url, setting="synchronize_config_to_ip")
+        config.set(value=username, setting="remote_system_username")
+        config.set(value=password, setting="remote_system_password")
+
+
+def plugins_xmlrpc_sync() -> Dict[str, str]:
+    """
+    Get all services on the firewall which can even be synced
+    """
+    # https://github.com/opnsense/core/blob/66c684b2c66d26000129bfb161c6cbafe4175dc8/src/etc/inc/plugins.inc#L355
+    php_requirements = ["/usr/local/etc/inc/plugins.inc"]
+    php_command = """
+                foreach (plugins_xmlrpc_sync() as $key => $item) {
+                    echo $key.','.$item['description'].'\n';
+                }
+                """
+
+    # run php function
+    result = opnsense_utils.run_command(
+        php_requirements=php_requirements,
+        command=php_command,
+    )
+
+    # check for stderr
+    if result.get("stderr"):
+        raise OPNSenseGetInterfacesError("error encountered while getting services")
+    allowed_services = dict(
+        service.split(",") for service in result.get("stdout_lines")
+    )
+    return allowed_services
+
+
+def services_to_synchronize(
```

Comment:
Some comments on what you are doing here would be nice.

Reply:
ee795d54e1f0faa36462301ca61f01796edc630a

does that help?